gcp-vertex-ai-mojo-scorer/src/main/java/ai/h2o/mojos/deploy/gcp/vertex/ai/controller/ModelsApiController.java

```diff
@@ -74,8 +87,47 @@ public static ScoreRequest getRestScoreRequest(
     request.setIdField(gcpRequest.getParameters().getIdField());
     request.setFields(gcpRequest.getParameters().getFields());
     
+    // Check if a pre-processing script was provided, if so, use it first
+    Map<String, String> env = System.getenv();
+    String preProccessingScript = env.getOrDefault("PREPROCESSING_SCRIPT_PATH", "");
+    
+    if (!preProccessingScript.isEmpty()) {
+      // Write data to file to be injested by preprocessing script
+      String fileName = UUID.randomUUID().toString() + ".json";
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        mapper.writeValue(new File("/tmp/" + fileName), gcpRequest);
+      } catch (IOException e) {
+        log.error("Failed writing JSON file: {}", e.getMessage());
+      }
+      
+      // Run preprocessing script on request data
+      try (FileReader fileReader = new FileReader("/tmp/" + fileName);
+          JsonReader reader = new JsonReader(fileReader)) {
+        ProcessBuilder processBuilder;
+        processBuilder = new ProcessBuilder("python", "/tmp/preprocessing_script.py", fileName);
+        processBuilder.redirectErrorStream(true);
+        Process process = processBuilder.start();
+        process.waitFor();
+        
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        gcpRequest = gson.fromJson(reader,
+            ai.h2o.mojos.deploy.common.rest.vertex.ai.model.ScoreRequest.class);
+      } catch (JsonSyntaxException e) {
+        log.error("Malformed JSON when reading from file: {}", e.getMessage());
+      } catch (Exception e) {
+        log.error("Unexpected error during data preprocessing step: {}", e.getMessage());
+      } finally {
+        try {
+          Files.deleteIfExists(Paths.get("/tmp" + fileName));
```

Comment:
I mean anything in a finally clause really shouldn't throw an error.... but ok. I think in this context it kind of makes sense.